File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/roleplay/job/JobLevelUpMessage.scala





















// Generated on 05/03/2013 10:34:44
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait JobLevelUpMessage extends MessageDeserializer {
  val messageId = 5656

  protected trait Message extends BaseMessage with Serializable {
    val messageId = JobLevelUpMessage.messageId

    val newLevel: Byte
    val jobsDescription: JobDescription.TypeImpl

    def serialize(buf: Buffer) {
      buf.writeByte(newLevel)
      jobsDescription.serialize(buf)
    }

  }

  protected def create(newLevel: Byte, jobsDescription: JobDescription.TypeImpl): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val newLevel = buf.readByte()
    val jobsDescription = JobDescription.deserialize(buf)

    create(newLevel, jobsDescription)
  }

}

object JobLevelUpMessage extends JobLevelUpMessage {
  case class MessageImpl(newLevel: Byte, jobsDescription: JobDescription.TypeImpl) extends Message
  type Target = MessageImpl

  override def create(newLevel: Byte, jobsDescription: JobDescription.TypeImpl) = MessageImpl(newLevel, jobsDescription)

  def apply(newLevel: Byte, jobsDescription: JobDescription.TypeImpl) = create(newLevel, jobsDescription)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/JobLevelUpMessage.java


// Generated on 05/08/2013 19:37:47
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class JobLevelUpMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5656;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public byte newLevel;
    public JobDescription jobsDescription;
    
    public JobLevelUpMessage() { }
    
    public JobLevelUpMessage(byte newLevel, JobDescription jobsDescription) {
        this.newLevel = newLevel;
        this.jobsDescription = jobsDescription;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeByte(newLevel);
        jobsDescription.serialize(buf);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        newLevel = buf.readByte();
        if (newLevel < 0)
            throw new RuntimeException("Forbidden value on newLevel = " + newLevel + ", it doesn't respect the following condition : newLevel < 0");
        jobsDescription = new JobDescription();
        jobsDescription.deserialize(buf);
    }
    
}
